testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -5820,3 +5819,313 @@ fn atlas_stress_integration_test() {
+    let bob_addr = to_addr(&bob_sk);
+    let bob_pd: PrincipalData = bob_addr.into();
+
+    let (mut conf, _) = neon_integration_test_conf();
+
+    // Create a schedule where we lower the read_count on Epoch2_05.
+    conf.burnchain.epochs = Some(vec![
+        StacksEpoch {
+            epoch_id: StacksEpochId::Epoch20,
+            start_height: 0,
+            end_height: 220,
+            block_limit: ExecutionCost {
+                write_length: 100000000,
+                write_count: 1000,
+                read_length: 1000000000,
+                read_count: 150,
+                runtime: 5000000000,
+            },
+        },
+        StacksEpoch {
+            epoch_id: StacksEpochId::Epoch2_05,
+            start_height: 220,
+            end_height: 9223372036854775807,
+            block_limit: ExecutionCost {
+                write_length: 100000000,
+                write_count: 1000,
+                read_length: 1000000000,
+                read_count: 50,
+                runtime: 5000000000,
+            },
+        },
+    ]);
+
+    conf.initial_balances.push(InitialBalance {
+        address: alice_pd.clone(),
+        amount: 10492300000,
+    });
+    conf.initial_balances.push(InitialBalance {
+        address: bob_pd.clone(),
+        amount: 10492300000,
+    });
+    conf.initial_balances.push(InitialBalance {
+        address: creator_pd.clone(),
+        amount: 10492300000,
+    });
+
+    test_observer::spawn();
+    conf.events_observers.push(EventObserverConfig {
+        endpoint: format!("localhost:{}", test_observer::EVENT_OBSERVER_PORT),
+        events_keys: vec![EventKeyType::AnyEvent],
+    });
+
+    let mut btcd_controller = BitcoinCoreController::new(conf.clone());
+    btcd_controller
+        .start_bitcoind()
+        .map_err(|_e| ())
+        .expect("Failed starting bitcoind");
+
+    let mut btc_regtest_controller = BitcoinRegtestController::new(conf.clone(), None);
+    let http_origin = format!("http://{}", &conf.node.rpc_bind);
+
+    btc_regtest_controller.bootstrap_chain(200);
+
+    let mut run_loop = neon::RunLoop::new(conf);
+    let blocks_processed = run_loop.get_blocks_processed_arc();
+    let burnchain_height = run_loop.get_burnchain_height_arc();
+
+    let channel = run_loop.get_coordinator_channel().unwrap();
+
+    thread::spawn(move || run_loop.start(None, 0));
+
+    // Wait until block 210, so we now that the burnchain is ready.
+    wait_for_runloop(&blocks_processed);
+    run_until_burnchain_height(
+        &mut btc_regtest_controller,
+        &blocks_processed,
+        &burnchain_height,
+        210,
+    );
+
+    // Publish the contract so we can use it.
+    submit_tx(
+        &http_origin,
+        &make_contract_publish(
+            &creator_sk,
+            0,
+            1100000,
+            "increment-contract",
+            &giant_contract,
+        ),
+    );
+
+    // Wait to make sure the contract is published.
```

Comment:
If this is unreliable, then you can simply query the node after each burnchain block to see if the contract was instantiated.

Reply:
these are possibilities but do you want to hold up the PR for this?

Reply:
Bounds have been tightened up on this test, so marking this thread resolved -- if there's still issues with this test, we should discuss them on the PR, not the commit (this test has moved to a different file entirely in the PR).